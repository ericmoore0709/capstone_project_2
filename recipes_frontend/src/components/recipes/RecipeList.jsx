import { CardGroup } from 'reactstrap';
import RecipeCard from './RecipeCard';
import PropTypes from 'prop-types';

const RecipeList = ({ title, recipes, deleteRecipe, shelfOptions, addRecipeToShelf }) => {
    return (
        <div>
            <h2 className='text-center mt-2'>{title}</h2>

            {/* Show message if there are no recipes to display */}
            {recipes.length === 0 && (
                <p className="text-center my-3">No recipes to display.</p>
            )}

            {/* Render recipes if available */}
            <CardGroup className='d-flex flex-wrap m-1 w-100 justify-content-center'>
                {recipes.map((recipe) => (
                    <RecipeCard
                        key={recipe.id}
                        recipe={recipe}
                        deleteRecipe={deleteRecipe}
                        shelfOptions={shelfOptions}
                        addRecipeToShelf={addRecipeToShelf}
                    />
                ))}
            </CardGroup>
        </div>
    );
}

RecipeList.propTypes = {
    title: PropTypes.string.isRequired,
    recipes: PropTypes.array.isRequired,
    deleteRecipe: PropTypes.func,
    shelfOptions: PropTypes.array,
    addRecipeToShelf: PropTypes.func
};

export default RecipeList;
