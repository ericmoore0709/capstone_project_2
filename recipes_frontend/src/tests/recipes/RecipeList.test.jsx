import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import RecipeList from "../../components/recipes/RecipeList";
import { MemoryRouter } from "react-router-dom";

describe('RecipeList', () => {

    const mockSignedInUser = { id: 1 };

    const renderComponent = (props) => {
        return render(
            <MemoryRouter>
                <RecipeList signedInUser={mockSignedInUser} deleteRecipe={vi.fn()} {...props} />
            </MemoryRouter>
        );
    };

    it('renders without crashing', () => {
        renderComponent({ recipes: [] });
    });

    it('takes a snapshot', () => {
        const { asFragment } = renderComponent({ recipes: [] });
        expect(asFragment()).toMatchSnapshot();
    });

    it('displays correct number of recipe cards', () => {
        const recipes = [
            { id: 1, title: 'Recipe 1', description: 'Tasty!', author_id: 1 },
            { id: 2, title: 'Recipe 2', description: 'Tastier!', author_id: 1 },
            { id: 3, title: 'Recipe 3', description: 'Tastiest!', author_id: 2 },
            { id: 4, title: 'Recipe 4', description: 'Somehow still tastier!', author_id: 2 }
        ];
        renderComponent({ recipes });
        expect(screen.getAllByRole('heading', { level: 5 })).toHaveLength(recipes.length);
    });

    it('displays no cards when recipes array is empty', () => {
        renderComponent({ recipes: [] });
        const recipeCards = screen.queryAllByRole('heading', { level: 5 });
        expect(recipeCards).toHaveLength(0);
    });

    it('correctly displays all recipe titles and descriptions', () => {
        const recipes = [
            { id: 1, title: 'Recipe 1', description: 'Tasty!', author_id: 1 },
            { id: 2, title: 'Recipe 2', description: 'Tastier!', author_id: 1 }
        ];
        renderComponent({ recipes });
        recipes.forEach((recipe) => {
            expect(screen.getByText(recipe.title)).toBeInTheDocument();
            expect(screen.getByText(recipe.description)).toBeInTheDocument();
        });
    });

    it('displays "No recipes to display." message when there are no recipes', () => {
        renderComponent({ recipes: [] });
        expect(screen.getByText("No recipes to display.")).toBeInTheDocument();
    });
});
