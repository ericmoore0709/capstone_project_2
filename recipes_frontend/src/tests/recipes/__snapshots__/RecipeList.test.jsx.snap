// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecipeList > takes a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-center mt-2"
    />
    <p
      class="text-center my-3"
    >
      No recipes to display.
    </p>
    <div
      class="d-flex flex-wrap m-1 w-100 justify-content-center card-group"
    />
  </div>
</DocumentFragment>
`;
